tests/analysis/impl/dense/gpcca_test.py

```diff
@@ -0,0 +1,468 @@
+
+        index = _indexsearch(sys)
+
+        np.testing.assert_array_equal(index, [0, 5, 2])
+
+    def test_initialize_A_shape_error_1(self):
+        X = np.zeros((3, 4))
+        X[:, 0] = 1.0
+        with pytest.raises(
+            ValueError,
+            match=r"The Schur vector matrix of shape \(\d+, \d+\) has more columns than rows. "
+            r"You can't get a \d+-dimensional simplex from \d+ data vectors.",
+        ):
+            _initialize_rot_matrix(X)
+
+    def test_initialize_A_first_is_not_constant(self):
+        X = np.zeros((4, 4))
+        X[0, 0] = 1.0
+        with pytest.raises(
+            ValueError,
+            match="First Schur vector is not constant. This indicates that the Schur vectors are incorrectly sorted. "
+            "Cannot search for a simplex structure in the data.",
+        ):
+            _initialize_rot_matrix(X)
+
+    def test_initialize_A_second_is_constant(self):
+        X = np.zeros((3, 3))
+        X[:, 0] = 1.0
+        with pytest.raises(
+            ValueError,
+            match="A Schur vector after the first one is constant. Probably the Schur vectors are incorrectly sorted. "
+            "Cannot search for a simplex structure in the data.",
+        ):
+            _initialize_rot_matrix(X)
+
+    def test_initialize_A_condition(self):
+        dummy = hilbert(14)
+        dummy = dummy[:, :-1]
+        dummy[:, 0] = 1.0
+
+        with pytest.raises(ValueError, match="The condition number .*"):
+            _initialize_rot_matrix(dummy)
+
+    def test_initialize_A(self):
+        mu0 = mu(0)
+        P, sd = get_known_input(mu0)
+        X, _, _ = _do_schur(P, sd, m=4)
+        evs = X[:, :4]
+
+        A = _initialize_rot_matrix(evs)
+        index = _indexsearch(evs)
+        A_exp = pinv(X[index, :4])
+
+        assert_allclose(A, A_exp)
+
+    def test_initialize_A_condition_warning(self):
+        dummy = hilbert(6)
+        dummy = dummy[:, :-1]
+        dummy[:, 0] = 1.0
+
+        with pytest.warns(UserWarning):
+            _ = _initialize_rot_matrix(dummy)
+
+    def test_objective_shape_error_1(self):
+        svecs = np.zeros((4, 3))
+        svecs[:, 0] = 1.0
+        alpha = np.zeros((9,))
+
+        with pytest.raises(
+            ValueError, match="The shape of alpha doesn't match with the shape of X: .+"
+        ):
+            _objective(alpha, svecs)
+
+    def test_objective_shape_error_2(self):
+        svecs = np.zeros((3, 4))
+        svecs[:, 0] = 1.0
+        alpha = np.zeros((4,))
+
+        with pytest.raises(
+            ValueError, match="The shape of alpha doesn't match with the shape of X: .+"
+        ):
+            _objective(alpha, svecs)
+
+    @pytest.mark.xfail(reason="No check whether 1st Schur vectors is 1")
+    def test_objective_1st_col(self):
+        svecs = np.zeros((3, 4))
+        alpha = np.zeros((9,))
+
+        with pytest.raises(ValueError):
+            _objective(alpha, svecs)
```

Comment:
@msmdev, this fails because there is an extra check for constant-1 first schur vector in Matlab that is not present in our implementation, because we test for it already in the function which calls the objective function.